src/presence_analyzer/utils.py

```diff
@@ -71,15 +72,27 @@ def get_data():
     return data
 
 
-def group_by_weekday(items):
+def group_by_weekday(items, *args, **kwargs):
     """
     Groups presence entries by weekday.
     """
+    if 'start' in kwargs and kwargs['start'] and \
+            'end' in kwargs and kwargs['end']:
```

Comment:
`'start' in kwargs` this is enough.
U should use default params IMO.

Anyway create a seaparate function. It will be more readable. It could return a tuple `(start_times, end_times)`